contract/src/Custody.sol

```diff
@@ -361,25 +381,70 @@ contract Custody is IChannel, IDeposit {
     }
 
     /**
-     * @notice Reset will close and open channel for resizing allocations
-     * @param channelId Unique identifier for the channel
-     * @param candidate The latest known valid state
-     * @param proofs An array of valid state required by the adjudicator
-     * @param newChannel New channel configuration (must have exactly 2 participants)
-     * @param newDeposit Initial State defined by the opener, containing the expected allocation
+     * @notice All participants agree in setting a new allocation resulting in locking or unlocking funds
+     * @param channelId Unique identifier for the channel to resize
+     * @param candidate The state with CHANRESIZE magic number and resize amounts
      */
-    function reset(
+    function resize(
         bytes32 channelId,
-        State calldata candidate,
-        State[] calldata proofs,
-        Channel calldata newChannel,
-        State calldata newDeposit
+        State calldata candidate
     ) external {
-        // First close the existing channel
-        close(channelId, candidate, proofs);
+        Metadata storage meta = _channels[channelId];
+
+        // Verify channel exists and is ACTIVE
+        if (meta.stage == Status.VOID) revert ChannelNotFound(channelId);
+        if (meta.stage != Status.ACTIVE) revert InvalidStatus();
+
+        // Verify all participants have signed the resize state
+        // For our 2-participant channels, we need exactly 2 signatures
+        if (candidate.sigs.length != 2) revert InvalidStateSignatures();
+        if (!_verifyAllSignatures(meta.chan, candidate)) revert InvalidStateSignatures();
+
+        // Decode the magic number and resize amounts
+        (uint32 magicNumber, int256[] memory resizeAmounts) = abi.decode(candidate.data, (uint32, int256[]));
+        if (magicNumber != CHANRESIZE || resizeAmounts.length != 2) revert InvalidState();
+
+        uint256 initialBalanceSum = meta.actualDeposits[0].amount + meta.actualDeposits[1].amount;
+        int256 finalBalanceSum = int256(initialBalanceSum) + resizeAmounts[0] + resizeAmounts[1];
+        if (finalBalanceSum < 0) revert InvalidState();
+        uint256 candidateBalanceSum = candidate.allocations[0].amount + candidate.allocations[1].amount;
+        if (uint256(finalBalanceSum) != candidateBalanceSum) revert InvalidState();
+
+        // Process resize amounts for each participant
+        for (uint256 i = 0; i < 2; i++) {
```

Comment:
Maybe it doesn't require a for loop, but a small helper inlined too, the code is a bit lengthy and we need more readable

Reply:
6ce4d7f44750d3fe041bfefb40dfc7e8fb52a0fb